lib/ddtrace/contrib/sidekiq/base_tracer.rb

```diff
@@ -0,0 +1,64 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/contrib/sidekiq/ext'
+
+module Datadog
+  module Contrib
+    module Sidekiq
+      class BaseTracer
```

Comment:
Could we turn this into a module instead of a class? (For composition vs inheritance?) We could probably name this something like `module Tracing` so the classes instead read:

```ruby
class ClientTracer
  include Tracing
end

class ServerTracer
  include Tracing
end
```